### PR TITLE
Support stat for dirs without files inside

### PR DESCRIPTION
**Pull request by jfontan**
It could not stat directories that contained only other dirs. For
example:

* dir
  * another_dir
    * a_file.txt

Trying to stat "/dir" didn't work.

### CODE REVIEW

**Comment by vmarkovtsev** [comment]
@mcuadros Are we good to go?

File: filesystem_test.go [hunk 1]
```diff
@@ -195,16 +195,21 @@ func (s *FilesystemSuite) TestReadFs(c *C) {
...
 	for _, path := range f.contents {
-		f, err := fs.Open(path)
+		s, err := fs.Stat(path)
 		c.Assert(err, IsNil)
-		c.Assert(f, NotNil)
 
-		read, err := ioutil.ReadAll(f)
-		c.Assert(err, IsNil)
-		c.Assert(len(read) > 0, Equals, true)
+		if !s.IsDir() {
```
Comment (smola) [inline filesystem_test.go]: Maybe there should be two different tests here.

Comment (mcuadros) [reply filesystem_test.go]: Yep

Comment (jfontan) [reply filesystem_test.go]: I'll add new tests for directories.

Comment (jfontan) [reply filesystem_test.go]: Added specific tests for nested dirs.

**Pull request merged**